### PR TITLE
feat: selective tool disabling

### PR DESCRIPTION
**Pull request by illya-havsiyevych**
addresses https://github.com/BeehiveInnovations/zen-mcp-server/issues/115

## Description

This PR adds support for selectively disabling tools via the DISABLED_TOOLS environment variable, allowing users to customize which MCP tools are available in their Zen server instance. This feature enables better control over tool availability for security,  performance, or organizational requirements.

## Changes Made

- [x] Added `DISABLED_TOOLS` environment variable support to selectively disable tools
- [x] Implemented tool filtering logic with protection for essential tools (version, listmodels)
- [x] Added comprehensive validation with warnings for unknown tools and attempts to disable essential tools
- [x] Updated `.env.example` with DISABLED_TOOLS documentation and examples
- [x] Added comprehensive test suite (16 tests) covering all edge cases
- [x] No breaking changes - feature is opt-in with default behavior unchanged

## Configuration

Add to `.env` file:
```bash
# Optional: Tool Selection
# Comma-separated list of tools to disable. If not set, all tools are enabled.
# Essential tools (version, listmodels) cannot be disabled.
# Available tools: chat, thinkdeep, planner, consensus, codereview, precommit,
#                  debug, docgen, analyze, refactor, tracer, testgen
# Examples:
# DISABLED_TOOLS=                    # All tools enabled (default)
# DISABLED_TOOLS=debug,tracer        # Disable debug and tracer tools
# DISABLED_TOOLS=planner,consensus   # Disable planning tools
```

### CODE REVIEW

File: tests/test_disabled_tools.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            ("chat", {"chat"}),  # Single tool
+            ("chat,debug", {"chat", "debug"}),  # Multiple tools
+            ("chat, debug, analyze", {"chat", "debug", "analyze"}),  # With spaces
+            ("chat,debug,chat", {"chat", "debug"}),  # Duplicates
+        ],
+    )
+    def test_parse_disabled_tools_parametrized(self, env_value, expected):
+        """Parametrized tests for various input formats."""
+        with patch.dict(os.environ, {"DISABLED_TOOLS": env_value}):
+            assert parse_disabled_tools_env() == expected
```
[the remark this answers is not in the document]
Comment (illya-havsiyevych) [reply tests/test_disabled_tools.py]: @gemini-code-assist check the latest code

**Comment by guidedways** [comment]
Thanks @illya-havsiyevych !

**Comment by erickreutz** [comment]
Confused on how this works. With claude code and using `uvx` with the github repo and adding my `DISABLED_TOOLS` to the `env` key of the mcp json but no tools are disabled. 

```json
    "zen": {
      "command": "sh",
      "args": [
        "-c",
        "exec $(which uvx || echo uvx) --from git+https://github.com/BeehiveInnovations/zen-mcp-server.git zen-mcp-server"
      ],
      "env": {
        "DISABLED_TOOLS": "analyze,refactor,testgen,secaudit,docgen,tracer,thinkdeep"
      }
    },
```
